src/rgw/rgw_rados.cc

```diff
@@ -3270,6 +3270,21 @@ int RGWRados::BucketShard::init(const rgw_bucket& _bucket,
   return 0;
 }
 
+int RGWRados::BucketShard::init(const RGWBucketInfo& bucket_info,
+                                const rgw_obj& obj)
+{
+  bucket = bucket_info.bucket;
+
+  int ret = store->open_bucket_index_shard(bucket_info, index_ctx, shard_id, &bucket_obj);
```

Comment:
@cbodley where is shard_id initialized?

Reply:
oof, thanks. this is calling the wrong overload of open_bucket_index_shard(). i'll rerun the multisite tests with num_shards > 1 to make sure i get this right